src/Microsoft.ML.Core/Utilities/ResourceManagerUtils.cs

```diff
@@ -311,6 +297,28 @@ private Exception DownloadResource(IHostEnvironment env, IChannel ch, WebClient
             }
         }
 
+        public bool IsRedirectToDefaultPage(Uri uri)
+        {
+            try
+            {
+                HttpWebRequest myHttpWebRequest = (HttpWebRequest)WebRequest.Create(uri.AbsoluteUri);
+                myHttpWebRequest.AllowAutoRedirect = false;
+                HttpWebResponse myHttpWebResponse = (HttpWebResponse)myHttpWebRequest.GetResponse();
+            }
+            catch (WebException e)
+            {
+                // Redirects to https://www.microsoft.com/en-us/?ref=aka
+                if (e.Message == "The remote server returned an error: (302) Moved Temporarily.")
```

Comment:
The message from web servers is not guaranteed to remain the same. Like we discussed earlier, you should be using http status codes. 
See [https://docs.microsoft.com/en-us/dotnet/api/system.net.httpwebresponse.statuscode?view=netframework-4.8#System_Net_HttpWebResponse_StatusCode](https://docs.microsoft.com/en-us/dotnet/api/system.net.httpwebresponse.statuscode?view=netframework-4.8#System_Net_HttpWebResponse_StatusCode)